### ISSUE TITLE
Violation is reported on @Override annotation line instead of method declaration line for EqualsHashCodeCheck

### ISSUE THREAD

**Issue by ZaheerAhmadDev** [checkstyle/checkstyle#20318]
I have read check documentation: https://checkstyle.sourceforge.io/checks/coding/equalshashcode.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="EqualsHashCode"/>
  </module>
</module>
```

```java
$ cat src/test.java
class Test {
    @Override
    public boolean equals(Object o) { // Violation Should be on this line.
        return false;
    }
}
```

```
$  java  -jar checkstyle-13.6.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] C:\Users\PMYLS\Desktop\pattern\src\Test.java:2:5: Definition of 'equals()' without corresponding definition of 'hashCode()'. [EqualsHashCode]
Audit done.
Checkstyle ends with 1 errors.
```

```
$  java  -jar checkstyle-13.5.0-all.jar -t  src/Test.java
COMPILATION_UNIT -> COMPILATION_UNIT [1:1]
`--CLASS_DEF -> CLASS_DEF [1:1]
    |--MODIFIERS -> MODIFIERS [1:1]
    |--LITERAL_CLASS -> class [1:1]
    |--IDENT -> Test [1:7]
    `--OBJBLOCK -> OBJBLOCK [1:12]
        |--LCURLY -> { [1:12]
        |--METHOD_DEF -> METHOD_DEF [2:5]
        |   |--MODIFIERS -> MODIFIERS [2:5]
        |   |   |--ANNOTATION -> ANNOTATION [2:5]
        |   |   |   |--AT -> @ [2:5]                                           // Violation is currently reported at here. 
        |   |   |   `--IDENT -> Override [2:6]
        |   |   `--LITERAL_PUBLIC -> public [3:5]
        |   |--TYPE -> TYPE [3:12]
        |   |   `--LITERAL_BOOLEAN -> boolean [3:12]
        |   |--IDENT -> equals [3:20]                           
        |   |--LPAREN -> ( [3:26]
        |   |--PARAMETERS -> PARAMETERS [3:27]
        |   |   `--PARAMETER_DEF -> PARAMETER_DEF [3:27]
        |   |       |--MODIFIERS -> MODIFIERS [3:27]
        |   |       |--TYPE -> TYPE [3:27]
        |   |       |   `--IDENT -> Object [3:27]
        |   |       `--IDENT -> o [3:34]
        |   |--RPAREN -> ) [3:35]
        |   `--SLIST -> { [3:37]
        |       |--LITERAL_RETURN -> return [4:9]
        |       |   |--EXPR -> EXPR [4:16]
        |       |   |   `--LITERAL_FALSE -> false [4:16]
        |       |   `--SEMI -> ; [4:21]
        |       `--RCURLY -> } [5:5]
        `--RCURLY -> } [6:1]
```

**Describe what you expect in detail.**
 The Violation need to be reported on `line 3` where the method is declared, not on  `line 2` where the `@Override` annotation is. 
 Without `@Override` the violation is correctly reported on the method declaration line.

**Comment by ZaheerAhmadDev** [comment]
@romani, please take a look when you get a chance.

**Comment by romani** [comment]
@ZaheerAhmadDev , 
violation is placed on `METHOD_DEF [2:5]` , because when Check deal with method it keep reference to this AST node.
and yes, it has the same line number as Annotation in this example.
but without annotation it will be same as visual location of method signature.

Placing violation on some node is very hard decision, but most Checks already do for METHOD_DEF.

`|--IDENT -> equals [3:20]` is just another opinion-ed decision. and not represent method 

consider:
```
    public
        boolean 
            equals(Object o) { // Violation Should be on this line.
        return false;
    }
```

and here violation will be located on method name and not where method signature is started.

If we ever decide to do this, it will be breaking compatibility change, as you already see in your PR, when you changes Xpath suppression. The same must do all users in the world, do you think it is good effort for them ? 
we do not make location of violation better, just change location

**Comment by ZaheerAhmadDev** [comment]
Fair point the multi-line example shows there no single correct line here so its not worth the compatibility break, thanks for walking through it.

**Comment by romani** [comment]
I am closing issue for now, but discussions can continue in case some body have good idea
